src/main/scala/components/memory/base-memory-components.scala

```diff
@@ -0,0 +1,43 @@
+// Base memory classes
+
+package dinocpu
+
+import chisel3._
+import chisel3.util.experimental.loadMemoryFromFile
+
+/**
+  * Base class for all modular backing memories. Simply declares the IO and the memory file.
+  */
+class BaseDualPortedMemory(size: Int, memfile: String) extends Module {
```

Comment:
I suppose so, but I observed that the `BaseBranchPredictor` module isn't abstract as well so I figured that maybe there was some intention for this. If this shouldn't be the case then I'll correct both the base memory modules and the base branch predictor accordingly.

Reply:
Right... Let's do this in a separate patch. I'd like to see this merged sooner rather than later.

For the branch predictor... I was on a deadline and couldn't figure out how to make the class abstract, IIRC. ;)